src/main/java/io/confluent/connect/elasticsearch/ElasticsearchClient.java

```diff
@@ -479,14 +481,23 @@ protected void handleResponse(BulkItemResponse response, DocWriteRequest<?> requ
    * @param response the failed response from ES
    */
   private void handleMalformedDocResponse(BulkItemResponse response) {
-    String errorMsg = String.format(
-        "Encountered an illegal document error -> Response status: '%s',\n"
-                + "Index: '%s',\n Document Id: '%s'. \n"
-                + "Ignoring and will not index record.",
-        response.getFailure().getStatus(),
-        response.getFailure().getIndex(),
-        response.getFailure().getId()
-    );
+    String errorMsg = "";
+    if(logSensitiveData) {
+      errorMsg = String.format(
```

Comment:
nit: Considering checks for `logSensitiveData` at multiple places, I feel we could create string constants for different messages and if possible re-use parts of messages where applicable

Reply:
done